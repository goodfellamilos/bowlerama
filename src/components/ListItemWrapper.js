import React from "react";
import PropTypes from "prop-types";
import { ListItem } from "material-ui";
import DeleteButton from "../components/DeleteButton";
import { LIST_ITEM_STYLE } from "../constants/materialUIStyles";

const ListItemWrapper = ({ player, onClick }) => {
  const onButtonClick = () => {
    onClick(player.id);
  };

  return (
    <ListItem
      disableTouchRipple={true}
      style={LIST_ITEM_STYLE}
      primaryText={player.name}
      rightIconButton={<DeleteButton onClick={onButtonClick} />}
    />
  );
};

ListItemWrapper.propTypes = {
  player: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    scores: PropTypes.array.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
};

ListItemWrapper.defaultProps = {
  onClick: (e) => e,
};

export default ListItemWrapper;
