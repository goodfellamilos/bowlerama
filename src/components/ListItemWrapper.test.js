import React from 'react';
import ListItemWrapper from './ListItemWrapper';
import { getMuiTheme, MuiThemeProvider } from 'material-ui/styles';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

// Some of material ui components has to be mocked in order to create component snapshots
jest.mock('material-ui/IconButton', () => 'IconButton');

describe('ListItemWrapper component', () => {
  it('renders without crashing component ListItemWrapper', () => {
    shallow(
      <ListItemWrapper
          player={{
            name: 'Player Number One',
            id: '_playerNumberOne_',
            scores: []
          }}
          onClick={e => e} />
    );
  });

  it('renders without crashing component ListItemWrapper', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={getMuiTheme()}>
        <ListItemWrapper
            player={{
              name: 'Player Number One',
              id: '_playerNumberOne_',
              scores: []
            }}
            onClick={e => e} />
      </MuiThemeProvider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
