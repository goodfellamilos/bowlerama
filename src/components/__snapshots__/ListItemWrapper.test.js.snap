// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemWrapper component creates a snapshot of a component ListItemWrapper 1`] = `
<div>
  <span
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    open={null}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "backgroundColor": null,
        "border": 10,
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "cursor": "default",
        "display": "block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 16,
        "fontWeight": "inherit",
        "lineHeight": "16px",
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "padding": 0,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      style={
        Object {
          "marginLeft": 0,
          "muiPrepared": true,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 56,
          "paddingTop": 16,
          "position": "relative",
        }
      }
    >
      <IconButton
        onClick={[Function]}
        style={
          Object {
            "position": "absolute",
            "right": "4px",
            "top": 0,
          }
        }
        tooltip="Remove player"
        tooltipPosition="top-left"
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </IconButton>
      <div
        style={
          Object {
            "muiPrepared": true,
          }
        }
      >
        Player Number One
      </div>
    </div>
  </span>
</div>
`;
